tardis/utilities/utils.py

```diff
@@ -66,6 +67,32 @@ def csv_parser(
             }
 
 
+KeyType = TypeVar("KeyType")
+
+
+def deep_update(
+    original_mapping: Mapping[KeyType, Any],
+    mapping_update: Mapping[KeyType, Any],
+) -> Mapping[KeyType, Any]:
+    from collections.abc import Mapping
+
+    updated_mapping = deepcopy(original_mapping)
+    for key, value in mapping_update.items():
+        if isinstance(value, Mapping):
```

Comment:
If you decide to only support `dict` (see above comment), check here explicitly:
```suggestion
        if isinstance(value, dict):
```

Reply:
I would prefer to use `MutableMapping`, since it has to merge `AttributeDicts` as well. Through, it inherits from `dict`.